pkg/sql/sem/tree/datum.go

```diff
@@ -1270,6 +1270,8 @@ func AsDBytes(e Expr) (DBytes, bool) {
 	switch t := e.(type) {
 	case *DBytes:
 		return *t, true
+	case *DString:
+		return DBytes(*t), true
```

Comment:
I don't think it's correct to do this - the contract of the `AsDFoo` methods seems to be to return a DFoo from an Expr if it is in fact a DFoo, and not to do any implicit casting. I don't think it's wise to add an implicit cast to just one of these.